tests/cli/test_flow_run.py

```diff
@@ -192,6 +192,24 @@ def test_ls_state_name_filter(
     )
 
 
+def test_ls_state_name_case_insensitive_filter(
+    scheduled_flow_run,
+    completed_flow_run,
+    running_flow_run,
+    late_flow_run,
+):
+    result = invoke_and_assert(
+        command=["flow-run", "ls", "--state", "late"],
```

Comment:
I think we should add a test for `--state-type`

Reply:
I have not made any changes to state-type because it was already giving a error message with valid and possible state-type

So, not added test case as well

Reply:
Ah I see. I think we should make the change to `state-type` so we can close the issue. 

I'm not getting an error with a valid state type passed in: 
```bash
❯ prefect flow-run ls --state-type SCHEDULED
                                            Flow Runs
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━━━━━┓
┃                                   ID ┃ Flow       ┃ Name               ┃ State     ┃ When      ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━━━━┩
│ 05874d1e-de4a-45f6-aa81-a821f8cbf16d │ tired-flow │ laughing-stork     │ SCHEDULED │ in 1 hour │
│ 21057066-fe73-424e-a1ed-7b85ef674b15 │ tired-flow │ loose-platypus     │ SCHEDULED │ in 1 hour │
│ 9702159c-d3e3-4bd9-b0e5-7e63182a3525 │ tired-flow │ blond-heron        │ SCHEDULED │ in 1 hour │
└──────────────────────────────────────┴────────────┴────────────────────┴───────────┴───────────┘
```

So I think we should be able to same case what is passed in.

Reply:
So if you try using

prefect flow-run ls --state-type Scheduled


It will throw an error with valid and possible state

Reply:
Right, that is the error the linked issue describes

Reply:
If you can make that update to `state-type` then we can close the issue fully once this is in.

Reply:
Added a suggestion for that here: https://github.com/PrefectHQ/prefect/pull/12302/files#r1529632960